vault/deprecated.md

```diff
@@ -0,0 +1,2 @@
+Upcoming in Vault 1.14, we will stop publishing official Dockerhub images and publish only our Verified Publisher images. Users of Docker images should pull from [hashicorp/vault](https://hub.docker.com/r/hashicorp/vault) instead of [vault](https://hub.docker.com/_/vault). Verified Publisher images can be found at https://hub.docker.com/r/hashicorp/vault.
+
```

Comment:
```suggestion
```

I think this is all the CI is complaining about :sweat_smile:

(I'm happy to take care of this change if you'd rather or that would be helpful :bow:)